httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/HttpByteArrayCacheEntrySerializer.java

```diff
@@ -328,7 +330,7 @@ public HttpCacheStorageEntry deserialize(final byte[] serializedObject) throws R
                     statusLine.getStatusCode(),
                     responseHeaders,
                     resource,
-                    variantMap
+                    !variants.isEmpty() ? variants : null
```

Comment:
Shouldn't we delegate everything to the internal constructor of `HttpCacheEntry`? I think its redundant.

Reply:
> Shouldn't we delegate everything to the internal constructor of `HttpCacheEntry`? I think its redundant.

@arturobernalg I am not sure I understand. You mean the `variants.isEmpty()` check?

Reply:
@ok2c my bad.  No change need it.  Sorry about that